docs/lua-records.rst

```diff
@@ -0,0 +1,366 @@
+the second set is tried.
+
+This configuration makes sense in the following context::
+
+    www    IN    LUA    CNAME   ( ";if(continent('EU')) then return 'west.powerdns.org' "
+                                  "else return 'usa.powerdns.org' end" )
+
+
+This sends queries that are geolocated to Europe to 'west.powerdns.org', and
+the rest to 'usa.powerdns.org'. The configuration for that name would then
+be::
+
+    usa    IN    LUA    A    ( "ifurlup('https://www.lua.org/',           "
+                               "{{'198.51.100.1'}, {'192.0.2.1', '192.0.2.2'}},  "
+                               "{stringmatch='Programming in Lua'})              " )
+
+Note that the sets of IP addresses have reversed order - visitors geolocated
+outside of Europe will hit 198.51.100.1 as long as it is available, and the
+192.0.2.1 and 192.0.2.2 servers as backup.
+
+Advanced topics
+---------------
+By default, LUA records are executed with 'return ' prefixed to them. This saves
+a lof of typing for common cases. To run actual Lua scripts, start a record with a ';'
+which indicates no 'return ' should be prepended.
+
+To keep records more concise and readable, configuration can be stored in
+separate records. The full example from above can also be written as::
+
+    config    IN    LUA    LUA ("settings={stringmatch='Programming in Lua'}  "
+                                "EUips={'192.0.2.1', '192.0.2.2'}             "
+                                "USAips={'198.51.100.1'}                      ")
+
+    www       IN    LUA    CNAME ( ";if(continent('EU')) then return 'west.powerdns.org' "
+                                   "else return 'usa.powerdns.org' end" )
+
+    usa       IN    LUA    A    ( ";include('config')                               "
+                                  "return ifurlup('https://www.lua.org/',        "
+                                  "{USAips, EUips}, settings)                    " )
+
+    west      IN    LUA    A    ( ";include('config')                               "
+                                  "return ifurlup('https://www.lua.org/',        "
+                                  "{EUips, USAips}, settings)                    " )
+
+
+
+Preset variables
+----------------
+
+LUA rules run within the same environment as described in
+:doc:`modes-of-operation`.
+
+The Lua snippets can query the following variables:
+
+``who``
+~~~~~~~
+IP address of requesting resolver
+
+
+``ecswho``
+~~~~~~~~~~~
+The EDNS Client Subnet, should one have been set on the query. Unset
+otherwise.
+
+``bestwho``
+~~~~~~~~~~~~
+In absence of ECS, this is set to the IP address of requesting resolver.
+Otherwise set to the network part of the EDNS Client Subnet supplied by the
+resolver.
+
+Functions available
+-------------------
+
+Record creation functions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. function:: ifportup(portnum, addresses[, options])
+
+  Simplistic test to see if an IP address listens on a certain port. Note that
```

Comment:
Should probably explicitly mention that this will test TCP. Could mention that it will attempt a connection and consider it "up" if the connection establishes - no data will be sent or read from the connection.